gnocchi/storage/ceph.py

```diff
@@ -201,13 +200,9 @@ def _pending_measures_to_process_count(self, metric_id):
         object_prefix = self.MEASURE_PREFIX + "_" + str(metric_id)
         return len(list(self._list_object_names_to_process(object_prefix)))
 
-    def list_metric_with_measures_to_process(self, size, part, full=False):
-        names = self._list_object_names_to_process()
-        if full:
-            objs_it = names
-        else:
-            objs_it = itertools.islice(names, size * part, size * (part + 1))
-        return set([name.split("_")[1] for name in objs_it])
+    def _list_metric_with_measures_to_process(self):
+        objs_it = self._list_object_names_to_process()
+        return sorted(set(name.split("_")[1] for name in objs_it))
```

Comment:
The sorting needs to occur based on datetime stamp or it just sorts based on metric uuid which will favor any uuids that "sort to the top by default"  this is especially harmful if those metric uuids exist for a very long time. (similar to my test with persisting instances and thus persisting metric uuids.)  

What about:
```python
return [name.split("_")[1] for name in sorted(objs_it, key=lambda x: x.split("_", 3)[-1])]
```
A further benefit of sorting by the incoming date timestamp is that the behavior becomes deterministic over the randomness of uuids.